src/MSBuild/XMake.cs

```diff
@@ -3133,7 +3133,7 @@ private static void ProcessFileLoggers(string[][] groupedFileLoggerParameters, L
                 LoggerVerbosity defaultFileLoggerVerbosity = LoggerVerbosity.Detailed;
                 fileLogger.Verbosity = defaultFileLoggerVerbosity;
 
-                if (cpuCount == 1)
+                if (cpuCount == 1 && Environment.GetEnvironmentVariable("MSBUILDNOINPROCNODE") != "1")
```

Comment:
Not sure I understand this. Can you elaborate?

Reply:
I've added a comment. This is fixing a minor pre-existing perf bug where we assumed that `cpuCount == 1` means that we won't be moving logs from another process.